airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -222,6 +231,13 @@ def _get_concurrent_state(
         )
 
     def observe(self, record: Record) -> None:
+        # Because observe writes to the most_recent_cursor_value_per_partition mapping,
+        # it is not thread-safe. However, this shouldn't lead to concurrency issues
+        # because observe() is only invoked on the main thread and the map is broken
```

Comment:
Isn't this because it's only observe on a PartitionReader thread and therefore there is only one thread updating the same key at a time?